### PR TITLE
support specify install disk for install-os

### PR DESCRIPTION
**Pull request by yyscamper**
- add a new option 'installDisk' in install-os job.
 - installDisk support both drive name (string) or drive index (number).
 - if use specify drive index, install-job will lookup the 'driveId' catalog to find the corresponding drive wwid.

@RackHD/corecommitters @iceiilin @WangWinson @pengz1 @sunnyqianzhang 

Below is the dependent PR:
https://github.com/RackHD/on-http/pull/58

### CODE REVIEW

**Comment by yyscamper** [comment]
@zyoung51

File: spec/lib/jobs/install-os-job-spec.js [hunk 1]
```diff
@@ -87,21 +93,138 @@ describe('Install OS Job', function () {
...
+                return job._convertInstallDisk();
+            }).then(function() {
+                expect(job.options.installDisk).to.equal('naa.rstuvw');
+            });
+        });
+
+        it('should set correct installDisk linux wwid', function() {
+            job.options.osType = 'linux';
+            return Promise.resolve().then(function() {
+                return job._convertInstallDisk();
```
Comment (benbp) [inline spec/lib/jobs/install-os-job-spec.js]: FYI, a nice way you can also do this is like:

`return expect(job._convertInstallDisk()).to.become('/dev/test1');`

File: lib/jobs/install-os.js [hunk 2]
```diff
@@ -148,24 +154,76 @@ function installOsJobFactory(
...
+        var disk = self.options.installDisk;
+
+        //If disk is empty then do nothing.
+        //If disk is string, it means user directly input the drive wwid, so don't need conversion.
+        if ((!disk && disk !== 0) || _.isString(disk)) {
+            return;
+        }
+
+        if (!_.isNumber(disk)) {
+            throw new Error('The installDisk format is not correct');
```
Comment (benbp) [inline lib/jobs/install-os.js]: Just a little nitpick: I generally like to normalize what comes out of promise returning functions, so doing a `Promise.reject(new Error(...))`.

**Comment by benbp** [comment]
Just some minor style comments, up to you if you care to address them.

:+1: regardless.

File: spec/lib/jobs/install-os-job-spec.js (hunk 1 above)
Comment (yyscamper) [inline spec/lib/jobs/install-os-job-spec.js]: @benbp , Your suggestion is good and much simpler than my design, However what I really want to verify is the job.options.installDisk is correctly converted after called function _convertInstallDisk, not the promise resolved value. 

Change _convertInstallDisk to return the job.options.installDisk will make the ".to.become" works, but I still need some logic to ensure the "options.installDisk" is also changed to the resolved value.

**Comment by anhou** [comment]
+1

**Pull request merged**